src/Microsoft.Azure.WebJobs.Host/Queues/DispatchProcessor.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Queues
+{
+    /// <summary>
+    /// wrapper around shared queue
+    /// </summary>
+    public class DispatchProcessor
+    {
+        private readonly SharedQueueHandler _sharedQueue;
+        private readonly string _functionId;
+        internal DispatchProcessor(SharedQueueHandler sharedQueue, string functionId)
+        {
+            _sharedQueue = sharedQueue;
+            _functionId = functionId;
+        }
+
+        /// <summary>
+        /// enqueue
+        /// </summary>
+        public async Task ProcessAsync(JObject message, CancellationToken cancellationToken)
```

Comment:
>ProcessAsync [](start = 26, length = 12)

Process implies we do the work right now. Can we rename to Enqueue?

Reply:
changed to "EnqueueAsync"